lib/display/r_raster.c

```diff
@@ -152,7 +155,28 @@ int D_open_driver(void)
 
     init();
 
-    return 0;
+    /* if GRASS_REGION is not defined, a display module can choose to render
+     * for the entire computational region or wait for the monitor to call it
+     * with GRASS_REGION set to the current display extent (what is shown) in
+     * the monitor; the monitor defines GRASS_REGION to the display extent
+     * dynamically, but it's the display module who is responsible to contrain
+     * its rendering within this region to save computational time; if the
+     * display module does not check the return value of D_open_driver(), it
+     * will have to render the same content twice (once by the command line and
```

Comment:
"render twice" - It seems to me that should never happen and the display modules should be required to work properly rather than opting in it.

Reply:
I completely agree with you. However, at the moment, not all display modules **display** (e.g., `d.info`). Obviously, with my first `exit()` method, this module and any other modules relying on it didn't work.

Maybe, it would be better to add a new argument to `D_open_driver(exit_on_no_region)` to not exit and use `D_open_driver(1)` in displaying modules and `D_open_driver(0)` in non-displaying ones.